incident_viewer_demo/src/LineChart/LineChart.jsx

```diff
@@ -638,79 +696,117 @@ class LineChart extends PureComponent {
    * Render the incident "good" and "bad" periods reference lines on the chart.
    */
   updateIncident() {
-    const { goodIncidentSeries, badIncidentSeries, incidentLineGenerator } = this.props;
+    const { goodIncidentSeries, badIncidentSeries, incidentLineGenerator, hasIncident } = this.props;
 
-    const goodIncidentSeriesArray = [{x: goodIncidentSeries.start, y: goodIncidentSeries.download_speed_mbps_median}, {x: goodIncidentSeries.end, y: goodIncidentSeries.download_speed_mbps_median} ];
-    const badIncidentSeriesArray = [{x: badIncidentSeries.start, y: badIncidentSeries.download_speed_mbps_median}, {x: badIncidentSeries.end, y: badIncidentSeries.download_speed_mbps_median} ];
-    
     this.goodIncidentLine.selectAll('*').remove();
     this.badIncidentLine.selectAll('*').remove();
 
     this.updateIncidentArrow();
+    this.updateIncidentShading();
 
-    const goodBinding = this.goodIncidentLine.selectAll('g').data(goodIncidentSeriesArray);
-    const badBinding = this.badIncidentLine.selectAll('g').data(badIncidentSeriesArray);
-
-    // ENTER
-    const goodEntering = goodBinding.enter()
-      .append('path')
-        .attr('d', incidentLineGenerator(goodIncidentSeriesArray));
-    const badEntering = badBinding.enter()
-      .append('path')
-        .attr('d', incidentLineGenerator(badIncidentSeriesArray));
+    if (hasIncident) {
+      const goodIncidentSeriesArray = [{x: goodIncidentSeries.start, y: goodIncidentSeries.download_speed_mbps_median}, {x: goodIncidentSeries.end, y: goodIncidentSeries.download_speed_mbps_median} ];
+      const badIncidentSeriesArray = [{x: badIncidentSeries.start, y: badIncidentSeries.download_speed_mbps_median}, {x: badIncidentSeries.end, y: badIncidentSeries.download_speed_mbps_median} ];
     
 
-    // ENTER + UPDATE
-    goodBinding.merge(goodEntering)
-      .transition()
-      .call(incidentLineGenerator);
-
-    badBinding.merge(badEntering)
-      .transition()
-      .call(incidentLineGenerator);
+      // LINES
+      this.goodIncidentLine.append('path')
+      .classed('good-incident-line', true)
+      .attr('d', incidentLineGenerator(goodIncidentSeriesArray))
+    
+    this.badIncidentLine.append('path')
+      .classed('bad-incident-line', true)
+      .attr('d', incidentLineGenerator(badIncidentSeriesArray))
+    }
+  }
 
-    // EXIT
-    goodBinding.exit().remove();
-    badBinding.exit().remove();
+  updateIncidentShading() {
```

Comment:
Please add a top of method comment + comments throughout so that people can understand what is going on here